### ISSUE TITLE
Using TTM with sparse Time Series Data

### ISSUE THREAD

**Issue by kiran-vj** [ibm-granite/granite-tsfm#212]
Hi Folks!

We've been wondering what would be the ideal way to deal with sparse time series data given the fixed length requirement. Should we pad with zeroes/duplicate initital/last events to reach max length.

Has there been any analysis around this?

**Comment by vijaye12** [comment]
yes. if you are planning to finetune the model, you can append zeros or do repeat of initial timepoints as you suggested and the model will learn the augmented pattern easily.

for zeroshot - repeat works slightly better compared to zeroshot. However - both these approachs might still have some drop in performance based on the datasets under consideration..

**Comment by mrlucasrib** [comment]
> yes. if you are planning to finetune the model, you can append zeros or do repeat of initial timepoints as you suggested and the model will learn the augmented pattern easily.
> 
> for zeroshot - repeat works slightly better compared to zeroshot. However - both these approachs might still have some drop in performance based on the datasets under consideration..


Hello, I plan to work with time series data that have relatively few time points (an average of 12) compared to the minimum context length required by most foundation models. As I’m starting my master’s, I aim to compare state-of-the-art (SOTA) foundation models using this dataset.

Is there a standard method to address or adapt datasets with fewer time points, especially for meeting the minimum context length? This applies not only to TTM but also to other SOTA approaches.

I’d greatly appreciate any guidance on this, including references to papers or resources that could help substantiate this approach.

**Comment by vg11072001** [comment]
@mrlucasrib you can check out this technique for generating augmentation using a combination of TSMixup and KernelSynth from the paper "Chronos: Learning the Language of Time Series."

- ![image](https://github.com/user-attachments/assets/cc7641d3-7596-4eb8-bfff-f1358d3849b3)

This type of real world dataset challenge when dealing with small dataset and finetuning when we couldn't cross the limit of context_lenght also can we design for TTM. Please suggest your views: @vijaye12 

This type of challenge with real-world datasets with lesser context. Can we design a solution for TTM in this context? Please share your thoughts: @vijaye12
